pkg/util/version.go

```diff
@@ -0,0 +1,86 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"context"
+	"errors"
+
+	dataplanev1 "github.com/openstack-k8s-operators/dataplane-operator/api/v1beta1"
+	"github.com/openstack-k8s-operators/lib-common/modules/common/helper"
+	openstackv1 "github.com/openstack-k8s-operators/openstack-operator/apis/core/v1beta1"
+)
+
+// GetVersion
+func GetVersion(ctx context.Context, helper *helper.Helper) (*openstackv1.OpenStackVersion, error) {
+	client := helper.GetClient()
+	log := helper.GetLogger()
+	var version *openstackv1.OpenStackVersion
+	versions := &openstackv1.OpenStackVersionList{}
+	if err := client.List(ctx, versions); err != nil {
```

Comment:
Is it intentional that we are not filtering to the namespace when looking up the OpenStackVersion object?

Reply:
no. I'll add it.